### DISCUSSION TITLE
What can we do to improve the designer interface?

### DISCUSSION THREAD

**Discussion by gbishop** [UNC-Project-Open-AAC/OS-DPI#223] [Ideas]
I have a few things that I really hate about it. I bet you do too. Please share them here so we can make it better.

**Comment by sofia-bg** [comment]
This doesn't help for aesthetics, but with patterns and our confusion around selectors we (a) were wanting to think about a different name for that, or if there was a way to number them? (b) just like in actions if when the patterns tab is open and I am scanning through if the "selector" that is being activated/used could be highlighted with each keystroke? 

We thought that might help people to learn/understand selectors and places where breakdowns might be occurring?

**Reply by aetkie** [reply]
That would be helpful. It can be confusing knowing which part of the design is controlled/connected to the Selector when there are more than 1.

**Reply by gbishop** [reply]
Good idea. I think I could highlight (somehow) all the elements that are selected. 

What about group by? Maybe it highlights the groups it creates in different colors? Or highlights the first group? What do you think?

**Reply by sofia-bg** [reply]
Oh that is interesting, so that as you start to declare things it shows you what would be included? I was also thinking the reverse, that as you scan through the UI that then it highlights in the programming what selector is being activated. But your suggestion might be more helpful for the actually designing stage.

**Comment by aetkie** [comment]
I wish that when you hovered over a component, a little dialog box would appear explaining what that component did/could do without having to go back and forth between OS-DPI and the wiki

**Reply by gbishop** [reply]
I like the idea but those popups could get really annoying if they appear automatically. I find such things often get in the way of what I'm trying to do. 
Maybe there is a spot you have to hover over? 
How does a keyboard user access this?

**Reply by sofia-bg** [reply]
Maybe an that can be clicked? 
![image](https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/92817124/bb6cabe1-c6b8-45bb-b9c6-108a577f49bf)

**Comment by aetkie** [comment]
We often have multiple designs open at the same time. It would be cool if tabs in the browser updated with the file name in it for faster selection. 

<img width="1237" alt="Screenshot 2023-09-21 at 10 15 09 AM" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/113473933/04ccb6b9-05fe-4001-a931-320132645534">

**Reply by gbishop** [reply]
Excellent suggestion and easy. In fact maybe it used to do that?

**Reply by sofia-bg** [reply]
I don't think I remember it ever doing that, but would be so helpful if it just auto-populated with the name

**Reply by gbishop** [reply]
I have pushed the change to production to make it display the page title.

**Reply by aetkie** [reply]
Yes! This works and I like it. Thanks Gary

**Comment by sofia-bg** [comment]
Oh and to Andrea's point, a faster way to offload designs. As I work I save so many iterations my imported to list to choose from is long. If there was a way to select multiple to offload at once that would be so helpful. Or a way to put them in folders/tree hierarchy so could collapse and effectively hide ones you aren't working between.

**Reply by gbishop** [reply]
Yes we need an Unload... menu entry that brings up a dialog.

**Reply by gbishop** [reply]
Try out the new Unload... menu entry. It seems to work for me. It will even close the tab if you unload something that happens to be open. 

I'm wondering if we even need the plain Unload entry?

**Reply by aetkie** [reply]
the plain Unload entry prompts you to save the open design. Exporting does the same thing. Seems like it's redundant.

**Reply by gbishop** [reply]
That was a safety mechanism to prevent accidental loss of your work. Hitting menu entries by mistake happens. With no confirmation your work would be gone.

The Unload... dialog provides that safety by requiring you to check a box and warning you at the same time.

**Comment by gbishop** [comment]
How about drag and drop for rearranging components in the layout?

**Reply by sofia-bg** [reply]
YES that would be very helpful. Also, sometimes I forget that to move up an entire action I actually need to click on the part of the action with the origin (at least it doesn't move up unless I do that, I think because rearranges internally otherwise?) I have wondered what about a little box or bullet that is at the beginning of the row by the origin that you click on so that you know you are wanting that action to be the one that moves, but could then be used as the point for drag and drop? Maybe for keyboard then you tab control to it and have a key you press with the up or down arrow? (i.e., shift + ↑  and  shift + ↓)

**Reply by gbishop** [reply]
That happens to me also. I hate the ambiguity. I need to do a better job of indicating the target of a menu item.

**Reply by aetkie** [reply]
drag and drop sounds like a great solution

**Reply by gbishop** [reply]
I have pushed a new version that enables Shift-ArrowUp and Shift-ArrowDown to move components. Give it a try and let me know what you think.

**Reply by aetkie** [reply]
Works perfectly!

**Reply by gbishop** [reply]
I've made a few more changes so you should see less of the cases where some internal part (such as a condition) moves instead of the component. Now it should prefer moving the component wherever you are focused inside it.
